File: app/src/main/java/com/solidscorpion/medic/MainActivityContract.java
package com.solidscorpion.medic;

import com.solidscorpion.medic.pojo.ModelMenuItem;

import java.util.List;

public interface MainActivityContract {

    interface View {
        void onMenuItemsLoaded(List<ModelMenuItem> items);
    }

    interface Presenter {
        void loadMenuItems();
    }
}

File: app/src/main/java/com/solidscorpion/medic/MainActivityContract.kt
package com.solidscorpion.medic

import com.solidscorpion.medic.pojo.BaseItem
import com.solidscorpion.medic.pojo.ModelMenuItem

interface MainActivityContract {

    interface View {
        fun onMenuItemsLoaded(items: List<ModelMenuItem>)
        fun showProgress()
        fun hideProgress()
        fun showResults(results: List<BaseItem>)
    }

    interface Presenter {
        fun loadMenuItems()
        fun performSearch(text: CharSequence, delay: Long)
        fun onStop()
    }
}
